maven-core/src/main/java/org/apache/maven/plugin/PluginParameterExpressionEvaluatorV4.java

```diff
@@ -343,13 +352,13 @@ else if ( expression.startsWith( "settings" ) )
 
                 if ( pathSeparator > 0 )
                 {
-                    String pathExpression = expression.substring( 1, pathSeparator );
+                    String pathExpression = expression.substring( 0, pathSeparator );
                     value = ReflectionValueExtractor.evaluate( pathExpression, session.getSettings() );
                     value = value + expression.substring( pathSeparator );
                 }
                 else
                 {
-                    value = ReflectionValueExtractor.evaluate( expression.substring( 1 ), session.getSettings() );
+                    value = ReflectionValueExtractor.evaluate( expression, session.getSettings() );
                 }
             }
             catch ( Exception e )
```

Comment:
Can you explain why there is now a shift in the string?

Reply:
Because I found out this is broken.  If the interpolated string is `${settings.foo}`, the `expression` is `settings.foo` and we pass `ettings.foo` to the `ReflectionValueExtractor`.  It's working because the call just ignore whatever is on the left of the first `.`

Reply:
I've extracted those changes in a different PR.